rag/advanced_rag/knowlege_compile/raptor.py

```
@@ -253,118 +115,80 @@ async def _embedding_encode(self, txt):
-        labels = np.zeros(n, dtype=int)
-        cluster_id = 0
-        for i in range(1, n):
-            if adj_sims[i - 1] >= threshold:
-                labels[i] = cluster_id
-            else:
-                cluster_id += 1
-                labels[i] = cluster_id
+        sorted_sims = np.sort(adj_sims)  # ascending
+
+        # Max clusters allowed by the ratio cap
+        max_clusters = max(1, int(round(n * self._clustering_ratio)))
```

Ruff (0.16.0)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)